### ISSUE TITLE
[Suggestions] - Level Caps

### ISSUE THREAD

**Issue by FLalejandro** [Ordalca/Nuzlocke#5]
Something which has increased in popularity for Nuzlockes is self-imposed Level Caps. This is done to increase the difficulty, and prevent players from sweeping the game by massively over-leveling.

Typically, the level cap increases whenever a player defeats a gym leader in the Main Games, and a similar concept could be done in Pixelmon, whereupon defeating a Gym Leader, a command could be run, or the Nuzlocke mod could employ a "check" to scan for gym badges.

 It'd be cool to have forced-level caps which can be toggled and customized in the nuzlocke.yml. It could look something like this.
 `level-cap: true/false`
`level-cap-intervals: 5, 10, 15`
where level-cap-intervals increase the level cap upon a condition being met.

**Comment by Ordalca** [comment]
Level caps are doable, but it'd be hard to decide on what the caps could be.  The gyms in pixelmon are a bit too end-game for them to work as level caps, since the gym leader is a legendary boss, I think, so will always be 40-50 levels above you.

**Comment by FLalejandro** [comment]
That's true, they are very tough bosses so it wouldn't be ideal for a level 15 level cap at the first gym haha.

In the Main Games, Gym Towns tend to be separated by a random amount of routes, which allows for people who Nuzlocke to get a couple of encounters before each gym. 

What if we "translate" routes to biomes, and add a level cap per each biome you've discovered? Ex: Your level cap increases by "X" amount of levels per each biome you discover.

I assume that since the mod detects if a Pokemon has already been caught in that specific biome, it can detect if the player has traveled into it as well.

There could also maybe be detection for the number of NPCs fought which determines the level cap as well. Ex: After defeating 10 wild NPCs, your level cap increases by 5 levels.
